CakePHP/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -134,6 +148,158 @@ protected function processTagSpacing(File $phpcsFile, int $stackPtr, int $commen
+                    return;
+                }
+
+                if ($tagNode->type instanceof UnionTypeNode) {
+                    $sortedTypeHint = $this->getSortedTypeHint($tagNode->type->types);
+                    if (strpos($tagContent, $sortedTypeHint) === 0) {
+                        continue;
+                    }
+
+                    if ($tagNode instanceof ParamTagValueNode) {
+                        $sortedContent = trim(sprintf(
+                            '%s %s%s %s',
+                            $sortedTypeHint,
+                            $tagNode->isVariadic ? '...' : '',
+                            $tagNode->parameterName,
+                            $tagNode->description
+                        ));
+                        $fix = $phpcsFile->addFixableWarning(
+                            '%s @param type hint is not properly sorted, expected "%s"',
+                            $tag,
+                            'UnsortedTypes',
+                            [$tagNode->parameterName, $sortedTypeHint]
+                        );
+                        if ($fix) {
+                            $phpcsFile->fixer->beginChangeset();
+                            $phpcsFile->fixer->replaceToken($tag + 2, $sortedContent);
+                            $phpcsFile->fixer->endChangeset();
+                        }
+                    } elseif ($tagNode instanceof ReturnTagValueNode) {
+                        $sortedContent = trim(sprintf(
+                            '%s %s',
+                            $sortedTypeHint,
+                            $tagNode->description
+                        ));
+                        $fix = $phpcsFile->addFixableWarning(
+                            '%s() @return type hint is not properly sorted, expected "%s"',
+                            $tag,
+                            'UnsortedTypes',
+                            [$phpcsFile->getDeclarationName($stackPtr), $sortedTypeHint]
+                        );
+                        if ($fix) {
+                            $phpcsFile->fixer->beginChangeset();
+                            $phpcsFile->fixer->replaceToken($tag + 2, $sortedContent);
+                            $phpcsFile->fixer->endChangeset();
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param array $types node types
+     * @return string
+     */
+    protected function getSortedTypeHint(array $types): string
+    {
+        static $shouldSort = [
+            '\\Closure',
+            '\\Traversable',
+            '\\ArrayAccess',
+            '\\Stringable',
+            'mixed',
+            'callable',
+            'resource',
+            'object',
+            'iterable',
+            'array',
+            'string',
+            'float',
+            'int',
+            'bool',
+            'true',
+            'false',
+            'null',
+            'void',
```

Comment:
Is void required in this list given that it can only be used by itself?

Reply:
`void` is allowed in union types such as the return type of actions. The default, undeclared return type is `mixed|void` according to php.

Reply:
Makes no sense, should have been null ;) But meeh.

Reply:
> void is allowed in union types such as the return type of actions.

You mean in docblock types right? Surely you can't have `void` in a union in concreate PHP type.

Reply:
> Makes no sense, should have been null ;) But meeh.

I think the `void` in docblocks type is to make static analyzers happy else they will whine about missing return.